client/src/store/ticket.js

```diff
@@ -0,0 +1,122 @@
+import { observable, action } from 'mobx';
+
+import axios from '../config';
+import { notification } from 'antd';
+
+class TicketStore {
+    @observable tickets = [];
+    @observable loading = false;
+
+    @action
+    getTickets = async (data) => {
+        try {
+            this.loading = true;
+            const resObj = await axios.post('/tickets', data);
+            if (resObj?.data?.success) {
+                this.tickets = resObj.data.data;
+                this.loading = false;
+            }
+        } catch (err) {
+            console.log(err);
+            const { response } = err;
+            this.loading = false;
+            notification.error({ message: response?.data?.message || "Something went wrong !!" })
+        }
+    }
+
+    addTicket = async (data) => {
+        try {
+            this.loading = true;
+            const resObj = await axios.post('/tickets/add', data);
+            console.log(resObj);
+            if (resObj?.data?.success) {
+                this.tickets = [...this.tickets, resObj.data.data]
+                this.loading = false;
+            }
+        } catch (err) {
+            console.log(err);
+            const { response } = err;
+            this.loading = false;
+            notification.error({ message: response?.data?.message || "Something went wrong !!" })
+        }
+    }
+
+    @action
+    deleteTicket = async ({ hotel, user, _id }) => {
+        try {
+            this.loading = true;
+            const resObj = await axios.delete('/tickets/delete', { data: { hotel, user, ticket: _id } });
+            if (resObj?.data?.success) {
+                this.loading = false;
+                const found = this.tickets.find(i => i._id === _id)
+                this.tickets = this.tickets.filter(item => item != found);
```

Comment:
each item of tickets will be object, so not sure if it could be compared with `!=`. 
also it's redundant code, we can simplify it with one line.
this.tickets = this.tickets.filter(item => item._id != _id);